service/internal/builder/pipelines_builder.go

```diff
@@ -221,6 +221,19 @@ func (pb *pipelinesBuilder) buildPipeline(ctx context.Context, pipelineID config
 		zap.String(components.ZapNameKey, pipelineID.String()))
 	pipelineLogger.Info("Pipeline was built.")
 
+	// Some consumers may not correctly implement the Capabilities,
+	// and ignore the next consumer when calculated the Capabilities.
+	// Because of this wrap the first consumer if any consumers in the pipeline
+	// mutate the data and the first says that it doesn't.
+	if tc != nil {
+		tc = capabilitiesTraces{Traces: tc, capabilities: consumer.Capabilities{MutatesData: mutatesConsumedData}}
+	}
+	if mc != nil {
+		mc = capabilitiesMetrics{Metrics: mc, capabilities: consumer.Capabilities{MutatesData: mutatesConsumedData}}
+	}
+	if lc != nil {
+		lc = capabilitiesLogs{Logs: lc, capabilities: consumer.Capabilities{MutatesData: mutatesConsumedData}}
```

Comment:
I see that this change is noop, but I just found out `mutatesConsumedData` from _any_ pipeline affects all of them. Interesting why do we want that?